Cds 75075 store manage argo config (#3213)
harness/developer-hub: docs/continuous-delivery/gitops/connect-and-manage/manage-argo-configs.md

   
   <figcaption>Figure 2: GitOps cluster.</figcaption>
   </figure>

4. Add a Harness GitOps application for the Argo CD configuration.
   1. Use the Harness GitOps repository added earlier.
   2. In **Path**, select or add the path to the Argo CD configurations.
   
   <figure>
   
   <docimage path={require('./static/c42483d43b3d2464994cb0085125ba7b3b0ed10f0b5f5b901e9d31a9b66e1157.png')} width="60%" height="60%" title="Click to view full size image" />  
   
   <figcaption>Figure 3: Path to Argo CD configs in GitOps application.</figcaption>
   </figure>

   :::note
   
   If the repository directory has subordinate directories, select **Directory Recurse**.

   <figure>
   
   <docimage path={require('./static/33cab4c8688ebf13b142bdd87f826916afced3e1fd9120566fa2118eb2e2f43c.png')} width="60%" height="60%" title="Click to view full size image" />  
   
   <figcaption>Figure 4: Directory Recurse option.</figcaption>
   </figure>

   :::

   3. Use the Harness GitOps cluster added earlier. In **Namespace**, provide the namespace in which the agent is installed.

> 5. Save the Harness GitOps application.

Reviewer: we need to sync the application as well if the sync policy is manual.

## Git Configuration Files

Most entities, such as ConfigMaps, repository credentials, and applications will work without requiring modifications. Some entities, namely clusters, repositories, and AppProject, need some Harness-specific values added to the existing Argo-provided templates.

### Sample repository configuration

> Following is a sample repository configuration:

Reviewer 2: ```suggestion
Here's a sample repository configuration:
```

Reviewer 2: "Following" is just a little awkward. Minor suggestion.

Author: I use it because it's grammatically correct, but I completely agree. I've never liked it!

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: my-https-anon-repo
  namespace: argo-config-git
  labels:
    argocd.argoproj.io/secret-type: repository
stringData:
  name: my-https-anon-repo
  project: default
  url: https://github.com/argoproj/argocd-example-apps
  type: git
```

### Sample cluster configuration

> Following is a sample cluster config:

Reviewer 2: ```suggestion
Here's a sample cluster config:
```

```yaml
apiVersion: v1
data:
  config: eyJiZWFyZXJUb2tlbiI6InRlc3QiLCJ0bHNDbGllbnRDb25maWciOnsiaW5zZWN1cmUiOnRydWV9fQ==
  server: aHR0cHM6Ly90ZXN0LmNvbQo=
stringData:
  name: argogitcluster1
  project: default
kind: Secret
metadata:
  annotations:
    managed-by: argocd.argoproj.io
  labels:
    argocd.argoproj.io/secret-type: cluster
  name: cluster-test1
  namespace: argo-config-git
type: Opaque
```

#### Additional fields for clusters and repositories

> Following are the additional fields required for clusters and repositories:

Reviewer 2: ```suggestion
Here are the additional fields required for clusters and repositories:
```

- `stringData.project` or `data.project`, whose value is the AppProject name. Provide this field only if you want the entity to reside within the scope of a Harness project or organization.

- `stringData.name` or `data.name`, whose value is the name of the entity in Harness.  

The remaining fields remain the same as in Argo CD.

### Sample Argo AppProject configuration

> Following is a sample configuration for the Argo CD AppProject:

Reviewer 2: ```suggestion
Here's a sample configuration for the Argo CD AppProject:
```

```yaml
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  labels:
    accountId: b.<harness-account-id>.e
    orgId: <harness-organization>
    projectId: <harness-project>
  finalizers:
    - resources-finalizer.argocd.argoproj.io
  name: default
  namespace: argo-config-git
spec:
  clusterResourceWhitelist:
  - group: '*'
    kind: '*'
  destinations:
  - namespace: '*'
    server: '*'
  sourceRepos:
  - '*'
```

#### Additional fields required for the Argo CD AppProject

Following are the additional fields required for the AppProject:

- `metadata.labels.accountId`, whose value is the Harness account identifier.

- `metadata.labels.orgId`, whose value is the Harness organization identifier.

- `metadata.labels.projectId`, whose value is the Harness project identifier.

The remaining fields remain the same as in Argo CD.

:::note

> By default, Harness creates an AppProject for you when you create the repository and cluster at the beginning of this procedure. The name of this AppProject is available on the agent details page. If you are creating a new AppProject, remember to add the mapping for the Argo project to the Harness project in the GitOps Agent, and then use the value of `metadata.name` in the project field of the entity configurations. The UI requires that information to show the entities.

Reviewer 2: ```suggestion
By default, Harness creates an AppProject for you when you create the repository and cluster at the beginning of this procedure. The name of this AppProject is available on the agent details page, under **Mapped Harness Project**. If you are creating a new AppProject, remember to add the [mapping for the Argo project to the Harness project in the GitOps Agent](/docs/continuous-delivery/gitops/connect-and-manage/multiple-argo-to-single-harness), and then use the value of `metadata.name` in the project field of the entity configurations. The UI requires that information to show the entities.
```

:::

You can use all other entity configurations without modifying them.

### Sample application

> Following is a sample for application:

Reviewer 2: ```suggestion
Here's a sample application:
```
